File: IT/1/Eccezioni/Esercizi/Python/esercizio9.py
# file esercizio9.py

from __future__ import print_function
if hasattr(__builtins__, 'raw_input'): input = raw_input

square = -1

while (square < 0):
	s = input("Scrivi un numero: ")
	try:
		n = int(s)
		square = n * n
		print("{} al quadrato = {}".format(n, n * n))

	except ValueError as e:
		print(e)

print("Fine del programma.")

File: IT/1/Eccezioni/Esercizi/Java/Esercizio9.java
/* file Esercizio9.java */

public class Esercizio9 {
	public static void main(String [] args) {
		int square = 0;

		while (square == 0) {
			String s = System.console().readLine("Scrivi un numero: ");

			try {
				int n = Integer.parseInt(s);
				square = n * n;
				System.out.println(n + " al quadrato = " + square);
			}
			catch (NumberFormatException e) {
				System.out.println(e.getMessage());
			}
		}

		System.out.println("Fine del programma.");
	}
}

